jlcarrascof/PF-Henry-Project
# Dashboard for Hotels and Rooms

# 🚀 Pull Request: Feature Enhancements and Bug Fixes

🌟 **Summary:**
This pull request brings exciting feature enhancements and squashes some pesky bugs we discussed during our chat! 🎉

## Changes Made:

### 1. 🏨 **HotelDashboard Component:**
   - Improved logic for disabling hotels. 🛠️
   - Separated logic for disabling hotels and rooms. 🏡
   - Added the 'Disabled' property for hotels and 'Availability' for rooms. 🔄

### 2. 🚪 **TableRow Component:**
   - Replaced `findDOMNode` to avoid issues in StrictMode. 🚧
   - Integrated a switch for each room to enable/disable availability. 💡

### 3. 🛠️ **EnhancedTableToolbar Component:**
   - Updated the delete button functionality to work with the `disableRoom` function. 🗑️
   - Ensured proper selection handling for disabling multiple rooms. ✅

### 4. 📊 **createData Function:**
   - Updated to include the 'Disabled' property for hotels. 🏢
   - Added 'Availability' property for rooms. 🔄

### 5. ⚡ **UseEffect Hook:**
   - Optimized `useEffect` to fetch hotel data only once to prevent excessive resource consumption. 🚀

### 6. 🎨 **Table Styles:**
   - Updated table styles to use light gray for better visual aesthetics. 🎨
   
![image](https://github.com/jlcarrascof/PF-Henry-Project/assets/129629916/7bb0803d-6cfc-45fc-9715-3822cd69bdec)


### 7. 🛠️ **Miscellaneous:**
   - Addressed TypeScript errors and improved code readability. 📘
   - Created a more modular structure for better maintainability. 🧩

## How to Test:
1. Ensure you have the latest code. 🔄
2. Run the application and navigate to the Hotels Dashboard. 🏨
3. Verify the functionality of disabling hotels and rooms. ✅
4. Check the appearance and behavior of the table. 🖥️

🔗 **Related Issues:**
- Fixes #146, #157 

🙏 **Thank you for your contribution!** 🤗
